### ISSUE TITLE
🐛 Bug: Audit and fix accessibility for browser reporter

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [mochajs/mocha#5129]
### Bug Report Checklist

- [X] I have read and agree to Mocha's [Code of Conduct](https://github.com/mochajs/mocha/blob/master/.github/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/mochajs/mocha/blob/master/.github/CONTRIBUTING.md)
- [X] I have searched for [related issues](https://github.com/mochajs/mocha/issues?q=is%3Aissue) and [issues with the `faq` label](https://github.com/mochajs/mocha/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Afaq%20), but none matched my issue.
- [X] I have 'smoke tested' the code to be tested by running it outside the real test suite to get a better sense of whether the problem is in the code under test, my usage of Mocha, or Mocha itself.
- [X] I want to provide a PR to resolve this

### Expected

There should be no [web accessibility issues](https://web.dev/explore/accessible) with the browser reporter.

### Actual

A quick [axe DevTools scan](https://www.deque.com/axe/devtools) reveals 3 [Elements must meet minimum color contrast ratio thresholds](https://dequeuniversity.com/rules/axe/4.8/color-contrast) issues.

### Minimal, Reproducible Example

See `index.html` from https://github.com/mochajs/mocha-examples/pull/72.

### Versions

Mocha 10.4.0.

### Additional Info

This has been an issue for a while. I wanted to wait until after #5113 to bring it up, so as to not get in the way of that PR. 🙂

**Comment by mark-wiemer** [comment]
Oh, I can look at this!

**Comment by mark-wiemer** [comment]
The short answer is that at Microsoft we use Playwright and axe-core for automated a11y testing. We'd have to know what browsers are supported (#5226). The other constraint is that axe-core doesn't simulate user experience, so we'd have to use Playwright to get the page into all the states we'd expect that might have an impact on user experience: all tests passing, some failing, all failing, pending, skipped, etc.

Once automated bugs are fixed (or at least logged) we'd look at manual testing via Accessibility Insights for Web. It's a browser extension that has a ~20-part assessment to go through WCAG 2.1 AA and guide testers through manually checking all parts of a website, including high contrast support, screen readers, and other experiences using assistive technology.

Just noting this for now, not sure if I'll have time to look at things this month, but I haven't forgotten about this one :)
